Python/bytecodes.c

```diff
@@ -4180,26 +4624,26 @@ dummy_func(
         }
 
         tier2 pure op(_LOAD_CONST_INLINE, (ptr/4 -- value)) {
-            value = Py_NewRef(ptr);
+            value = PyStackRef_FromPyObjectNew(ptr);
         }
 
         tier2 pure op(_LOAD_CONST_INLINE_BORROW, (ptr/4 -- value)) {
-            value = ptr;
+            value = PyStackRef_FromPyObjectSteal(ptr);
```

Comment:
```suggestion
            value = PyStackRef_FromPyObjectImmortal(ptr);
```

Reply:
It seems this change isn't right. With this I get segfaults on the code because the assertion `_Py_IsImmortal()` fails. It seems when the executor is being invalidated, this assertion can fail for some reason?

Reply:
Think there's a bug in tier 2 actually.

Reply:
`PyStackRef_FromPyObjectSteal` is wrong and `PyStackRef_FromPyObjectImmortal` is correct, so please make this change.

What's the assertion failure?

Reply:
```
python: ./Include/internal/pycore_stackref.h:191: PyStackRef_FromPyObjectImmortal: Assertion `_Py_IsImmortal(obj)' failed.
Fatal Python error: Aborted
```
Triggered in `test_capi.test_opt`.

Reply:
Alright done. Fixed in https://github.com/python/cpython/pull/118450/commits/30f7a09f86f00dcaa6c3e60f99becf470d78c740 thanks to Sam.